### PR TITLE
Fixes for EC2 module integration tests.

### PR DESCRIPTION
**Pull request by mattclay**
##### SUMMARY

Fixes for EC2 module integration tests:

- ec2_elb_lb
- ec2_group
- ec2_key
- ecs_ecr

##### ISSUE TYPE

Bugfix Pull Request

##### COMPONENT NAME

EC2 integration tests

##### ANSIBLE VERSION

```
ansible 2.4.0 (aws-test-fixes 6b3933c2bf) last updated 2017/05/03 12:52:49 (GMT +800)
  config file = 
  configured module search path = [u'/Users/mclay/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/mclay/code/mattclay/ansible/lib/ansible
  executable location = /Users/mclay/code/mattclay/ansible/bin/ansible
  python version = 2.7.11 (default, Jan 22 2016, 08:29:18) [GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)]
```

##### ADDITIONAL INFORMATION

The following changes were made:

- Added missing block/always cleanup to reduce resource leakage.
- Added missing module credential parameters.
- Added support for running tests with AWS temporary security credentials.
- Split out SSH key pair generation for use only with tests which require it.
- Removed awk dependency for SSH key pair generation.
- Removed unused random string generation.
- Added missing module parameters.
- Corrected assertions.

These tests now pass with the latest boto versions installed:

- boto (2.46.1)
- boto3 (1.4.4)
- botocore (1.5.45)

Tests may fail with much older versions of boto due to differences in error messages.

The credentials template has been updated to add `security_token`. If you have an existing `credentials.yml` it may need to be updated.

### CODE REVIEW

**Comment by mattclay** [comment]
Due to the indentation changes from adding block/always, it will be much easier to view the diff if changes in spaces are ignored: https://github.com/ansible/ansible/pull/24231/files?w=1

**Review by gundalow — APPROVED** [review]
Looks sensible, though you may wish to get a nod from one of the cloud people.

File: test/integration/roles/ec2_provision_instances/meta/main.yml [hunk 1]
```diff
@@ -1,3 +1,4 @@
 dependencies:
   - prepare_tests
+  - setup_sshkey
   - setup_ec2
```
Comment (s-hertel) [inline test/integration/roles/ec2_provision_instances/meta/main.yml:4]: If setup_ec2 -> setup_sshkey, do we need to keep setup_ec2 as a dependency?

**Comment by s-hertel** [comment]
Somewhat related to https://github.com/ansible/ansible/pull/22499.

File: test/integration/roles/ec2_provision_instances/meta/main.yml (hunk 1 above)
Comment (mattclay) [reply test/integration/roles/ec2_provision_instances/meta/main.yml:4]: `setup_ec2` still has `defaults` and `tasks`, it just no longer has `tasks/main.yml`.

**Comment by mattclay** [comment]
Fixes still need to be made to other AWS integration tests. However, I limited my changes to these four module tests since they are the first ones that will be supported with the new AWS integration testing on Shippable. I just needed a few tests to pass for the initial roll out.

**Comment by s-hertel** [comment]
Looks great. shipit

**Review by s-hertel — APPROVED** [review]

**Pull request merged**